### ISSUE TITLE
Writing to CoreWebView2SharedBuffer times out

### ISSUE THREAD

**Issue by david-risney** [MicrosoftEdge/WebView2Feedback#3108]
_Originally posted by @Take-A-Byte in https://github.com/MicrosoftEdge/WebView2Feedback/issues/5#issuecomment-1371869026_
Moving to its own new issue.

> Also, Are these usable in C# - UWP? Because with the following code, it is not able to close the `writer` stream
```
           var baseStr = "Checking if small stream works";
            using (var strstream = GenerateStreamFromString(baseStr))
            {
                var environment = await CoreWebView2Environment.CreateAsync();
                using (CoreWebView2SharedBuffer sharedBuffer = environment.CreateSharedBuffer((ulong)strstream.Length))
                {
                    using (var stream = sharedBuffer.OpenStream())
                    {
                        using (StreamWriter writer = new StreamWriter(stream.AsStreamForWrite()))
                        {
                            writer.Write(strstream);
                        }
                    }
                }
            }
```

> It times out. Here is the exact error message that I get on the screen -

![image](https://user-images.githubusercontent.com/2541336/211068742-14a7c825-bf88-444b-a7ed-278727c7ebe6.png)

(Text extracted from image)
Exception User-Unhandled
System.lO.lOException: 'The message filter indicated that the application is busy.
A COM call (IID: method index: 5) to an ASTA (thread
18420) appears deadlocked and was timed out'
Inner Exception
Exception: The message filter indicated that the application is busy
A COM call (IID {0000000C-0000-0000-C000-000000000046}, method index: 5) to an ASTA (thread
18420) appears deadlocked and was timed out.

[AB#42876135](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/42876135)

**Comment by LiangTheDev** [comment]
@Take-A-Byte, while we investigate it, could you try to use alternative ways to work around the issue by using the following code instead of `StreamWriter(stream.AsStreamForWrite())`?

```
using (var writer = new Windows.Storage.Streams.DataWriter(stream))
{
  writer.WriteString(baseStr);
  await writer.StoreAsync();
}
```

**Comment by Take-A-Byte** [comment]
@LiangTheDev I tried this out and it works fine. Thanks for the suggestion 🙇 

Though I am trying to find what is the difference between StreamWriter and DataWriter and what are guidelines on which to use when but I could not find anything in Microsoft docs so far. If you know any place I can look for the answer or if you could shed some light on this topic, I would love to know about it :)

**Comment by Take-A-Byte** [comment]
@LiangTheDev Thanks for the explanation. 🙇

**Comment by champnic** [comment]
Fixed in runtimes 111.0.1615.0+. Thanks!
